optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -157,6 +163,20 @@ public void exceptionInConsumer() throws InterruptedException {
         assertEquals(SolverStatus.NOT_SOLVING, solverJob1.getSolverStatus());
     }
 
+    @Test
+    public void solveGenerics() {
+        final SolverConfig solverConfig = PlannerTestUtils
+                .buildSolverConfig(TestdataSolution.class, TestdataEntity.class);
+        SolverManager<TestdataSolution, Long> solverManager = SolverManager
+                .create(solverConfig, new SolverManagerConfig());
+
+        BiConsumer<Object, Object> exceptionHandler = (o1, o2) -> fail("This is unexpected");
+        Consumer<Object> finalBestSolutionConsumer = o -> {};
+        Function<Object, TestdataUnannotatedExtendedSolution> problemFinder = o -> new TestdataUnannotatedExtendedSolution();
+
+        solverManager.solve(1L, problemFinder, finalBestSolutionConsumer, exceptionHandler);
```

Comment:
This solver actually failed, but that failure never reached the test thread. I believe this change fixed it, please review :)
https://github.com/kiegroup/optaplanner/commit/d2589ed10073618ec7cbea7bfe0ce6cac55393ae